### PR TITLE
Let 352 fe project creation form

### PR DESCRIPTION
**Pull request by barbara-chaves**
Project creation form - Base and first form page

This PR adds the Projects/new page, with the project form first page

### Acceptance criteria
JUST FIRST FORM PAGE

There will be a way for a signed up user with role “project developer” to create a project

The user is able to enter the [project’s mandatory and optional fields](https://docs.google.com/spreadsheets/d/1qtP-qkm8iS6OlluiJwRvQoi6rCns9iKhRYoNzc9qkGg/edit#gid=1884416048) (see “project” fields in the “forms” sheet)

There will be a way to validate the information entered in the fields

Errors are displayed if the information is incorrect 

There will be a way for the user to know that the content should be written in the language of the account

There will be a way confirm or dismiss the creation of the project

Note: Save/cancel

The layout must comply with the designs defined [HeCo - Work in progress](https://www.figma.com/proto/KC49uMR9t1YARmFR0Sr6aN/HeCo---Work-in-progress?node-id=1702%3A24167&viewport=396%2C48%2C0.06&scaling=scale-down&starting-point-node-id=1702%3A24167&show-proto-sidebar=1)

## Tracking

[LET-352](https://vizzuality.atlassian.net/browse/LET-352)

### CODE REVIEW

File: frontend/containers/project-form-pages/pages/general-information.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+                errorText={(errors?.involved_project_developer as FieldError)?.message}
+              />
+            </fieldset>
+            {/* Project developers selector */}
+            <div
+              className={cx('relative top-[-66px] left-16 w-[360px]', {
+                hidden: !showInvolvedProjectDevelopers,
+              })}
+            >
+              <Combobox
```
Comment (clementprdhomme) [inline frontend/containers/project-form-pages/pages/general-information.tsx:242]: 🟠

You can use the new `<MultiCombobox />` for this now. It was merged very recently. Be careful, the value needs to be an array and you need to use React Hook Forms' `NestedValue` type (check the examples on Storybook).

File: frontend/containers/project-form-pages/pages/general-information.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+                }}
+                placeholder={formatMessage({
+                  defaultMessage: 'select project developers',
+                  id: 'QESbGa',
+                })}
+              >
+                {[
+                  ...projectDevelopers,
+                  // The hardcoded option Not Listed
+                  { id: 'not-listed', attributes: { name: 'not listed' } },
```
Comment (clementprdhomme) [inline frontend/containers/project-form-pages/pages/general-information.tsx:261]: 🟠

You need to translate the option. Also, could the first letter be uppercase (“Not listed”)? Thanks!

File: frontend/schemas/project.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { useIntl } from 'react-intl';
+
+import { object, string, array, number, mixed, boolean } from 'yup';
+
+export default (page: number) => {
+  const { formatMessage } = useIntl();
+
+  const messages = {
+    project_gallery: {
+      min_lenth: formatMessage({ defaultMessage: 'Upload at least one picture', id: 'BG1UW6' }),
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:10]: 🟠

This field is optional so there shouldn't be any minimum number of pictures.

File: frontend/schemas/project.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      max_length: formatMessage({ defaultMessage: 'Upload a maximum of six images', id: 'U6/vVg' }),
+      max_pitcure_size: formatMessage({
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:12]: 🟢

Small typo:
```suggestion
      max_picture_size: formatMessage({
```

File: frontend/schemas/project.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        defaultMessage: 'The pictures must have a maximun size of 5 Mbs',
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:13]: 🟠

Megabytes not megabits 😉 (+ small typo):
```suggestion
        defaultMessage: 'The pictures must have a maximum size of 5 MB',
```

File: frontend/schemas/project.ts (hunk 3 above)
Comment (clementprdhomme) [inline frontend/schemas/project.ts:10]: 🟢

```suggestion
      min_length: formatMessage({ defaultMessage: 'Upload at least one picture', id: 'BG1UW6' }),
```

File: frontend/schemas/project.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  const schemas = [
+    object().shape({
+      name: string().required(),
+      country_id: string().required(),
+      municipality_id: string().required(),
+      department_id: string().required(),
+      project_gallery: mixed<FileList>()
+        .test('min_lenth', messages.project_gallery.min_lenth, (value) => value?.length > 0)
+        .test('max_lenth', messages.project_gallery.max_length, (value) => value?.length <= 6)
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:31]: 🟢

```suggestion
        .test('max_length', messages.project_gallery.max_length, (value) => value?.length <= 6)
```

File: frontend/schemas/project.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  const schemas = [
+    object().shape({
+      name: string().required(),
+      country_id: string().required(),
+      municipality_id: string().required(),
+      department_id: string().required(),
+      project_gallery: mixed<FileList>()
+        .test('min_lenth', messages.project_gallery.min_lenth, (value) => value?.length > 0)
```
Comment (clementprdhomme) [inline frontend/schemas/project.ts:30]: 🟠

```suggestion
```

**Review by clementprdhomme — COMMENTED** [review]
In addition to my comments below, I believe we could do 2 others improvements:
- The error messages are not very user friendly at the moment (see screenshot)
- The “select project developers” combobox doesn't show any error if non is selected

First page of the form with error messages

File: frontend/containers/project-form-pages/pages/general-information.tsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+  register,
+  errors,
+  control,
+  getValues,
+}: ProjectFormPagesProps<ProjectForm>) => {
+  const [showInvolvedProjectDevelopers, setShowInvolvedProjectDevelopers] = useState(false);
+  const { formatMessage } = useIntl();
+  const { locations } = useGroupedLocations();
+  // I don't know if there is a better way to get all items, since the default page size is 10. I'm assuming that there will not be more than 1000 locations.
+  const { projectDevelopers } = useProjectDevelopersList({ 'page[size]': 1000 });
```
[the remark this answers is not in the document]
Comment (barbara-chaves) [reply frontend/containers/project-form-pages/pages/general-information.tsx:30]: I just need the name, I will add that to the params

**Review by clementprdhomme — APPROVED** [review]
Great job!

**Pull request merged**